tiledb/sm/filesystem/s3.cc

```diff
@@ -535,32 +534,49 @@ Status S3::is_empty_bucket(const URI& bucket, bool* is_empty) const {
   return Status::Ok();
 }
 
-bool S3::is_bucket(const URI& bucket) const {
+Status S3::is_bucket(const URI& uri, bool* const exists) const {
   init_client();
 
-  if (!bucket.is_s3()) {
-    return false;
+  if (!uri.is_s3()) {
+    return LOG_STATUS(Status::S3Error(
+        std::string("URI is not an S3 URI: " + uri.to_string())));
   }
 
-  Aws::Http::URI aws_uri = bucket.c_str();
+  Aws::Http::URI aws_uri = uri.c_str();
   Aws::S3::Model::HeadBucketRequest head_bucket_request;
   head_bucket_request.SetBucket(aws_uri.GetAuthority());
   auto head_bucket_outcome = client_->HeadBucket(head_bucket_request);
-  return head_bucket_outcome.IsSuccess();
+  *exists = head_bucket_outcome.IsSuccess();
```

Comment:
@Shelnutt2 @ihnorton 

The intent of this patch is that we will be able to refactor this line into something like:
```
if (head_bucket_outcome.IsSuccess()) {
  *exists = true;
} else {
  if (head_bucket_outcome.Error == /* HTTP 404 */) {
    *exists = false;
  } else {
    return Status::S3Error("Connection failed");
  }
}
```